rpa-home/modules/ROOT/pages/automation-usecase-legacysystemdataextraction-extractfromscreen.adoc

```diff
@@ -0,0 +1,97 @@
+
+image::usecase-legacysystem-screeninteraction-run.png[]
+
+## Enter Text Into a Screen Element and Click a Button
+
+To search for the tracking number, you have to enter the order ID in the search field and click the search button.
+
+To find the search field, you need to find the word `Order ID` next to it on the application screen first. You use the *Search Pattern* Action Step to capture and compare the pattern:
+
+image::usecase-legacysystem-screeninteraction-capturepattern.png["When you capture a pattern, the screen freezes. You draw a box around the pattern and the search area on the frozen screen."]
+
+image::usecase-legacysystem-screeninteraction-searchpattern.png[]
+
+Finding the pattern returns a coordinate that you use in the next step to determine where to input the order ID. You use the *Mouse Action* Action Step to click into the search field next to the label `Order ID`:
+
+image:usecase-legacysystem-screeninteraction-mouseaction.png["The coordinates from the found pattern are used in the Relative To field of the Mouse Action Action Step."]
+
+The search field now has the focus, so you can just enter the order ID with a simulated keyboard. Instead of recording key strokes, you use the Activity Parameter `order_id` with the *Enter String* Action Step to copy the Order ID into the field:
+
+image::usecase-legacysystem-screeninteraction-enterstring.png[]
+
+You use *Mouse Action* again to start the search. To determine the position of the search button to click, you reuse the location of the `Order id` pattern to record the click position.
+
+image::usecase-legacysystem-screeninteraction-mouseaction-clickbutton.png[]
+
+## Check the Results
+
+You know that the search is successful when there is a result listed. As you can see on xref:automation-usecase-legacysystemdataextraction.adoc#trackingsystem-allentries-image[the screen], the tracking number should be listed in the column of the same name, so you search for that column first, using `Tracking No.` as the search pattern:
+
+image::usecase-legacysystem-screeninteraction-searchpattern-text.png[]
+
+Because it might be that the order has no corresponding tracking number in the system and you don't want to falsely return the initial value of the variable as tracking number, you check this case first using the *Managed* Action Step.
+
+In the *DoAction*, which works like a try element, you check if the result list is empty. As you can see on xref:automation-usecase-legacysystemdataextraction.adoc#trackingsystem-noentries-image[the screen], in case of no results, an up arrow is shown directly below the empty space where the results should have been.
```

Comment:
```suggestion
In *DoAction*, which works like a try element, you check if the result list is empty. As you can see on xref:automation-usecase-legacysystemdataextraction.adoc#trackingsystem-noentries-image[the screen], in case of no results, an up arrow is shown directly below the empty space where the results should have been.
```